src/components/SignUp/EduForm.js

```diff
@@ -108,62 +56,253 @@ class EduForm extends Component {
     this.props.onChangeText('concentration', 'name', this.props.id, text);
   }
 
-  onChangeStartYear(year) {
-    this.props.onChangeText('startYear', 'name', this.props.id, year);
-    this.setState({ startYear: year });
+  onChangeDegree() {
+
+  }
+
+  onChangeStartYear(data) {
+    const year = data.label;
+    const date = year + '-' + this.state.startDate.month;
+    this.props.onChangeText('start_date', '', this.props.id, year);
+    this.state.startDate.year = year;
+    this.setState({});
+  }
+
+  onChangeStartMonth(data) {
+    const month = data.label;
+    const monthStr = month.length === 1 ? '0' + month : month;
+    const date = this.state.startDate.year + '-' + monthStr;
+    this.props.onChangeText('start_date', '', this.props.id, date);
+    this.state.startDate.month = monthStr;
+    this.setState({});
```

Comment:
you might wanna initialize all the data variables in the constructor in order to use setState.